### PR TITLE
Make the request from the libevent dispatch thread.

### PR DESCRIPTION
**Pull request by pphaneuf**

### CODE REVIEW

File: cpp/net/url_fetcher.cc [hunk 1]
```diff
@@ -186,8 +198,10 @@ void UrlFetcher::Post(const Request& req, Response* resp, Task* task) {
   TaskHold hold(task);
 
   State* const state(
-      new State(&impl_->pool_, EVHTTP_REQ_GET, req, resp, task));
+      new State(&impl_->pool_, EVHTTP_REQ_POST, req, resp, task));
```
Comment (pphaneuf) [inline cpp/net/url_fetcher.cc]: Oops! :smiley:

File: cpp/net/url_fetcher.cc [hunk 2]
```diff
@@ -40,9 +42,13 @@ struct State {
     CHECK(!conn_) << "request state object still had a connection at cleanup?";
   }
 
+  // The following methods are only called on the libevent dispatch
```
Comment (AlCutter) [inline cpp/net/url_fetcher.cc]: /are only/should (must?) only be/ ?

Comment (AlCutter) [inline cpp/net/url_fetcher.cc]: Also, there's `Base::OnEventThread()` if you want to ensure it's so (or just log if it isn't).

Comment (pphaneuf) [inline cpp/net/url_fetcher.cc]: Updated the comment and used ```Base::OnEventThread()```, as suggested.

**Comment by AlCutter** [comment]
LGTM

**Pull request merged**